src/FastConv/FastConv.cpp

```diff
@@ -12,21 +12,190 @@ CFastConv::~CFastConv( void )
+
+        m_pfFreqInput = new CFft::complex_t[fftBlockSize];
+        m_pfFreqIr = new CFft::complex_t[fftBlockSize];
+        m_pfFreqConv = new CFft::complex_t[fftBlockSize];
+
+        m_pfTimeInput = new float[fftBlockSize];
+        m_pfTimeIr = new float[fftBlockSize];
+
+        m_pfRealInput = new float[fftBlockSize];
+        m_pfImagInput = new float[fftBlockSize];
+        m_pfRealIr = new float[fftBlockSize];
+        m_pfImagIr = new float[fftBlockSize];
+        m_pfRealConv = new float[fftBlockSize];
+        m_pfImagConv = new float[fftBlockSize];
+
+        m_pfTmpConv = new float[fftBlockSize];
+
+        for (int i = 0; i < iLengthOfBuffers; i++) 
+        {
+            pfOutputBuffer[i] = 0;
+        }
+
+        int nBlockInput = 0;
+        while ((static_cast<float>(nBlockInput) < (static_cast<float>(iLengthOfBuffers) / static_cast<float>(m_iblockSize))) || (nBlockInput == 0 ))
+        {
+            for (int i = m_iblockSize * nBlockInput; i < m_iblockSize * (nBlockInput + 2); i++)
+            {
+                if(nBlockInput == 0) m_pfTimeInput[i] = (i < m_iblockSize || (i - m_iblockSize) > iLengthOfBuffers - 1) ? 0 : pfInputBuffer[i - m_iblockSize];
+                else m_pfTimeInput[i] = (i < m_iblockSize * (nBlockInput + 1) && i < iLengthOfBuffers) ? pfInputBuffer[i] : 0;
+            }
+            int nBlockIr = 0;
+            while ((static_cast<float>(nBlockIr) < (static_cast<float>(m_iLengthOfIr) / static_cast<float>(m_iblockSize))) || (nBlockIr == 0 ))
+            {
+                for (int i = m_iblockSize * nBlockIr; i < m_iblockSize * (nBlockIr + 2); i++)
+                {
+                    if (nBlockIr == 0) m_pfTimeIr[i] = (i < m_iblockSize || (i - m_iblockSize) > m_iLengthOfIr - 1) ? 0 : m_pfImpulseResponse[i - m_iblockSize];
+                    else m_pfTimeIr[i] = (i < m_iblockSize * (nBlockIr + 1) && i < m_iLengthOfIr) ? m_pfImpulseResponse[i] : 0;
+                }
+                m_pCFftInstance->doFft(m_pfFreqInput, m_pfTimeInput);
+                m_pCFftInstance->doFft(m_pfFreqIr, m_pfTimeIr);
+
+                for (int i = 0; i < fftBlockSize; i++)
+                {
+                    m_pfFreqInput[i] *= fftBlockSize;
+                    m_pfFreqIr[i] *= fftBlockSize;
+                }
+                m_pCFftInstance->splitRealImag(m_pfRealInput, m_pfImagInput, m_pfFreqInput);
+                m_pCFftInstance->splitRealImag(m_pfRealIr, m_pfImagIr, m_pfFreqIr);
+
+                for (int i = 0; i < (fftBlockSize); i++)
+                {
+                    m_pfRealConv[i] = (m_pfRealInput[i] * m_pfRealIr[i]) - (m_pfImagInput[i] * m_pfImagIr[i]);
+                    m_pfImagConv[i] = (m_pfRealInput[i] * m_pfImagIr[i]) + (m_pfImagInput[i] * m_pfRealIr[i]);
+                }
+                m_pCFftInstance->mergeRealImag(m_pfFreqConv, m_pfRealConv, m_pfImagConv);
+                m_pCFftInstance->doInvFft(m_pfTmpConv, m_pfFreqConv);
+
+                if (m_bLongBlock)
```

Comment:
for readability and staying concise, you may want to replace this with if (m_iblockSize > iLengthOfBuffers)